frappe/utils/nestedset.py

```diff
@@ -237,6 +237,28 @@ def validate_loop(doctype, name, lft, rgt):
 		frappe.throw(_("Item cannot be added to its own descendents"), NestedSetRecursionError)
 
 
+def remove_subtree(doctype: str, name: str, throw=True):
+	"""Remove doc and all its children."""
+	frappe.has_permission(doctype, ptype="delete", throw=throw)
+
+	# Determine the `lft` and `rgt` of the subtree to be removed.
+	lft, rgt = frappe.db.get_value(doctype, name, ["lft", "rgt"])
+
+	# Delete the subtree by removing all nodes whose values for `lft` and `rgt`
+	# lie within above values or match them.
+	frappe.db.delete(doctype, {"lft": (">=", lft), "rgt": ("<=", rgt)})
```

Comment:
ermm... this bypasses on_trash hooks and other things that happen when you do `delete_doc` 😅

Reply:
>  On the downside, it's on the database level and, thus, doesn't check for existing links.

Correct, this is just a developer utility, as many others. If you're using this, you'll need to take care of any other required logic yourself.

Reply:
lets add this hint in docstring at least.